server/app/repository/ExportServiceRepository.java

```diff
@@ -19,41 +18,42 @@ public ExportServiceRepository() {
   /**
    * Queries for all unique option admin names ordered by their creation time.
    *
-   * @param (@link QuestionDefinition) of a Checkbox question
-   * @throws (@link RuntimeException) when the questionDefinition is not of type Checkbox
+   * @param questionDefinition {@link QuestionDefinition} of a Checkbox question
+   * @throws RuntimeException when the questionDefinition is not of type Checkbox
    */
   public ImmutableList<String> getAllHistoricMultiOptionAdminNames(
       QuestionDefinition questionDefinition) {
     if (!questionDefinition.getQuestionType().isMultiOptionType()) {
       throw new RuntimeException("The Question Type is not a multi-option type");
     }
     String questionName = questionDefinition.getName();
-    LinkedHashSet<String> allOptions = new LinkedHashSet<>();
-    database
-        .sqlQuery(
-            "SELECT"
-                + "  all_options.admin_name "
-                + "FROM ( "
-                + "  SELECT "
-                + "    q.create_time, "
-                + "    jsonb_array_elements(q.question_options) ->>'adminName' AS admin_name, "
-                + "    jsonb_array_elements(q.question_options) ->>'id' AS id "
-                + "  FROM questions q "
-                + "  INNER JOIN versions_questions vq ON  vq.questions_id = q.id "
-                + "  INNER JOIN versions v ON vq.versions_id = v.id "
-                + "  WHERE lifecycle_stage IN ('obsolete', 'active') "
-                + "  AND name = :currentQuestion "
-                + ") AS all_options "
-                + "GROUP BY all_options.admin_name "
-                + "ORDER BY MIN(all_options.create_time), MIN(all_options.id) ASC")
-        .setParameter("currentQuestion", questionName)
-        .findList()
-        .stream()
-        .forEachOrdered(sqlRow -> allOptions.add(sqlRow.getString("admin_name")));
+    ImmutableList<String> allOptions =
```

Comment:
Can we switch back to linkedhashset to preserve order every time?

Reply:
Since the query has a GROUP BY we're already guaranteed to have a set of unique admin names, so we don't need to use the `LinkedHashSet` to de-dupe, and `ImmutableList` also preserves order.